server/src/github.com/alexeyknyshev/cpsrv/cpsrv.go

```diff
@@ -130,47 +111,20 @@ func getRequestJsonStr(r *http.Request, context string) (string, error) {
 	return string(jsonStr), nil
 }
 
-func prepareResponse(w http.ResponseWriter, r *http.Request) (bool, int64) {
-	requestId, err := getRequestUserId(r)
-	if err != nil {
-		log.Printf("%s prepareResponse %v\n", getRequestContexString(r), err)
-		w.WriteHeader(http.StatusBadRequest)
-		return false, 0
-	}
-	if requestId == 0 {
-		log.Printf("%s prepareResponse unexpected requestId: %d\n", getRequestContexString(r), requestId)
-		w.WriteHeader(http.StatusBadRequest)
-		return false, 0
-	}
-
-	w.Header().Set("Content-Type", "application/json; charset=utf-8")
-	w.Header().Set("Id", strconv.FormatInt(requestId, 10))
-	return true, requestId
-}
-
-func writeResponse(w http.ResponseWriter, r *http.Request, requestId int64, responseBody string) {
-	io.WriteString(w, responseBody)
-	go logResponse(w, r, requestId, responseBody)
-}
-
-func writeHeader(w http.ResponseWriter, r *http.Request, requestId int64, code int) {
-	w.WriteHeader(code)
-	go logResponse(w, r, requestId, "code " + strconv.FormatInt(int64(code), 10))
-}
-
 type EndpointCallback func(w http.ResponseWriter, r *http.Request)
 
-func handlerPing(tnt *tarantool.Connection) (string, EndpointCallback) {
+func handlerPing(handlerContext HandlerContext) (string, EndpointCallback) {
 	return "/ping", func(w http.ResponseWriter, r *http.Request) {
-		ok, requestId := prepareResponse(w, r)
+		logger := handlerContext.logger()
```

Comment:
It's copying Logger struct here. In case it would contain some non-trival data inside itself it will copy it again and again for each handlerCall which will be quite frequently on a production server). This thing **must** be fixed in **all** handlers.

P.S. As you can find in web everyone with high-load net apps trying to make as minimum as possible allocations in handlers or move them to async calls using goroutines (approach used by yours logger for example)